### PR TITLE
Adding dropdown widget

### PR DESCRIPTION
**Pull request by charlesx2013**

### CODE REVIEW

File: app/assets/javascripts/authentication/components/TestingPage.js [hunk 1]
```diff
@@ -85,15 +87,39 @@ class TestingPage extends React.Component {
       sections: this.state.sections
     });
   }
+
+  onBlur1(value) {
```
Comment (warrenshen) [inline app/assets/javascripts/authentication/components/TestingPage.js]: I am not a fan of these names.

Comment (warrenshen) [inline app/assets/javascripts/authentication/components/TestingPage.js]: Where do you use these functions?

File: app/assets/javascripts/shared/components/widgets/Dropdown.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  }
+}
+
+Dropdown.propTypes = {
+  header: PropTypes.element.isRequired,
+  items: PropTypes.array.isRequired,
+}
+
+Dropdown.defaultProps = {
+  items: [],
```
Comment (warrenshen) [inline app/assets/javascripts/shared/components/widgets/Dropdown.js]: You don't need a default prop if you make it required.

File: app/assets/javascripts/authentication/components/TestingPage.js (hunk 1 above)
Comment (charlesx2013) [inline app/assets/javascripts/authentication/components/TestingPage.js]: this is the testing page. not used for much else.

**Pull request merged**